### DISCUSSION TITLE
Interpolation

### DISCUSSION THREAD

**Discussion by Laul** [patrykandpatrick/vico#608] [Questions]
Hey ! 
Sorry in advance, it can be a stupid question, but I can't find the solution by myself and I'd need your help.

Basically I just wanna have a "smooth" chart rather than lines between 2 points. I checked the point connector and played with different floats but nothing makes sense. 

Is there anything I'd missed ?  Do I need to set something additional ? 

Or, maybe it is because of the number of points , but not fully logical in my mind. Is there a number of values not to exceed to have proper interpolation? 

Below is what I currently have. 

![Screenshot_2024-03-10-20-55-23-068_com.laul.trackaid-edit.jpg](https://github.com/patrykandpatrick/vico/assets/2975027/f1fbfe9a-e733-4379-b040-b6febd8b6756)

Thanks a lot for the help!:)

**Comment by patrickmichalik** [comment]
Hello! The point count is the issue here. The `PointConnector` in use determines how the points are connected, but the line still has to go through each point for the chart to be accurate. There isn’t a specific threshold, but the narrower the curves are, the more they’ll resemble straight lines. To make the line smoother, you’d have to reduce the point density. For example, you could divide the list of _y_ values into smaller lists of a set length, apply an aggregate function to each of these lists, and give the chart the resulting values. You could also use a more advanced downsampling algorithm. It depends on what kind of result you’re looking for. Let me know if you have any questions.
